src/network/mcpe/protocol/types/event/EntityInteractEventData.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\network\mcpe\protocol\types\event;
+
+use pocketmine\network\mcpe\protocol\EventPacket;
+use pocketmine\network\mcpe\serializer\NetworkBinaryStream;
+
+final class EntityInteractEventData implements EventData{
+	/** @var int */
+	public $mobType;
+	/** @var int */
+	public $interactionType;
+	/** @var int */
+	public $mobVariant;
```

Comment:
are you sure about this field order? It doesn't look consistent with the others.

Reply:
I admit honestly that I did not quite understand their order.

It was about this order that the following facts informed me:

[MinecraftEventing::fireEventEntityInteracted(Player *, int, MinecraftEventing::InteractionType, int, unsigned char)](https://gist.github.com/Minejong/ccbd78120c2da2e2d7c48c70d07d0e4c)

This is from EventPacket::EventPacket(Player const*, Actor const*, MinecraftEventing::InteractionType):
![2019-10-14_19-39-26](https://user-images.githubusercontent.com/55767127/66768076-6a85eb80-eeba-11e9-902e-d342984dadac.png)